spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/LogisticRegressionClassifier.java

```diff
@@ -0,0 +1,147 @@
+      super(fileSetName, path, featureFieldsToInclude, featureFieldsToExclude, predictionField);
+      this.numFeatures = numFeatures;
+    }
+  }
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    Schema inputSchema = stageConfigurer.getInputSchema();
+
+    Preconditions.checkArgument(inputSchema != null, "Input Schema must be a known constant.");
+    config.validateConfigParameters(inputSchema, config.featureFieldsToInclude, config.featureFieldsToExclude,
+                                    config.predictionField, null);
+
+    outputSchema = SparkUtils.getOutputSchema(inputSchema, config.predictionField);
+    stageConfigurer.setOutputSchema(outputSchema);
+  }
+
+  @Override
+  public void initialize(SparkExecutionPluginContext context) throws Exception {
+    FileSet fileSet = context.getDataset(config.fileSetName);
+    Location modelLocation = fileSet.getBaseLocation().append(config.path);
+    if (!modelLocation.exists()) {
+      throw new IllegalArgumentException("Failed to find model to use for classification." +
+                                           " Location does not exist: " + modelLocation, null);
+    }
+
+    // load the model from a file in the model fileset
+    JavaSparkContext javaSparkContext = context.getSparkContext();
+    SparkContext sparkContext = JavaSparkContext.toSparkContext(javaSparkContext);
+
+    String modelPath = modelLocation.toURI().getPath();
+    loadedModel = LogisticRegressionModel.load(sparkContext, modelPath);
+    tf = new HashingTF((config.numFeatures));
+  }
+
+  @Override
+  public JavaRDD<StructuredRecord> transform(SparkExecutionPluginContext context,
+                                             JavaRDD<StructuredRecord> input) throws Exception {
+    if (!input.isEmpty()) {
+      final Schema inputSchema = input.first().getSchema();
+      final Map<String, Integer> featuresList = SparkUtils.getFeatureList(inputSchema,
+                                                                          config.featureFieldsToInclude,
+                                                                          config.featureFieldsToExclude,
+                                                                          config.predictionField);
+
+      final JavaRDD<StructuredRecord> output = input.map(new Function<StructuredRecord, StructuredRecord>() {
+        @Override
+        public StructuredRecord call(StructuredRecord structuredRecord) throws Exception {
+          List<Object> result = new ArrayList<>();
+          for (String column : featuresList.keySet()) {
+            if (structuredRecord.get(column) != null) {
+              result.add(structuredRecord.get(column));
+            }
+          }
+          double prediction = loadedModel.predict(tf.transform(result));
```

Comment:
should not be using hashing tf. Feature generation should happen before this stage.